pkg/api/validation/schema.go

```diff
@@ -77,9 +77,15 @@ func (s *SwaggerSchema) ValidateBytes(data []byte) error {
 	if !ok {
 		return fmt.Errorf("error in unmarshaling data %s", string(data))
 	}
-	apiVersion := fields["apiVersion"].(string)
-	kind := fields["kind"].(string)
-	return s.ValidateObject(obj, apiVersion, "", apiVersion+"."+kind)
+	apiVersion := fields["apiVersion"]
+	if apiVersion == nil {
+		fmt.Errorf("apiVersion not set")
+	}
+	kind := fields["kind"]
+	if kind == nil {
+		fmt.Errorf("kind not set")
+	}
```

Comment:
err := json.Unmarshal(data, &obj), this already parses json package, I think we can't add an new unmarshal call, so I don't use runtime.UnstructuredJSONScheme.DataVersionAndKind. what's your point?